artiq/gateware/rtio/phy/urukul_monitor.py

```diff
@@ -0,0 +1,133 @@
+from migen import *
+
+from artiq.coredevice.spi2 import SPI_CONFIG_ADDR, SPI_DATA_ADDR, SPI_END, SPI_INPUT
+from artiq.coredevice.urukul import CS_DDS_CH0, CS_DDS_MULTI
+
+
+class AD99XXMonitorGeneric(Module):
+    def __init__(self, spi_rtlink):
+        self.rtlink = spi_rtlink
+
+        self.current_address = Signal.like(self.rtlink.o.address)
+        self.current_data = Signal.like(self.rtlink.o.data)
+        self.sync.rio += If(
+            self.rtlink.o.stb,
+            self.current_address.eq(self.rtlink.o.address),
+            self.current_data.eq(self.rtlink.o.data)
+        )
+
+        self.cs = Signal(8)
+        self.length = Signal(8)
+        self.flags = Signal(8)
+
+        self.sync.rio += If(
+            self.rtlink.o.stb & (self.current_address == SPI_CONFIG_ADDR),
+            self.cs.eq(self.current_data[24:]),
+            # self.div.eq(self.current_data[16:24] + 2),
+            self.length.eq(self.current_data[8:16] + 1),
+            self.flags.eq(self.current_data[0:8])
+        )
+
+    def selected(self, c):
+        if self.cs == CS_DDS_MULTI:
+            return True
```

Comment:
I suspect this doesn't do what you expect. ``if`` is always executed in the interpreter. You may need ``If``.